From 72cea6375e044a753abb171ef5f4bae2aabdd4d3 Mon Sep 17 00:00:00 2001
From: Saikrishna Arcot <sarcot@microsoft.com>
Date: Mon, 8 Jul 2024 17:09:00 -0700
Subject: [PATCH 1/2] Update disable-non-manylinux.patch due to pip being
 updated to 24.1.2.

The code logic in this area has changed, resulting in patch conflicts.

Signed-off-by: Saikrishna Arcot <sarcot@microsoft.com>
---
 sonic-slave-bookworm/disable-non-manylinux.patch | 16 +++++++++++-----
 sonic-slave-bullseye/disable-non-manylinux.patch | 16 +++++++++++-----
 2 files changed, 22 insertions(+), 10 deletions(-)

diff --git a/sonic-slave-bookworm/disable-non-manylinux.patch b/sonic-slave-bookworm/disable-non-manylinux.patch
index 67e81c4e9116..dcdb953581c9 100644
--- a/sonic-slave-bookworm/disable-non-manylinux.patch
+++ b/sonic-slave-bookworm/disable-non-manylinux.patch
@@ -1,10 +1,16 @@
+Disable any type of wheel besides manylinux wheels. This is to work around an issue where the architecture-specific
+wheel might be built with a newer glibc version than what is supported on the system.
+
+(This patch gets applied only on armhf.)
+
 --- a/tags.py	2022-07-12 00:07:22.710207780 +0000
 +++ b/tags.py	2022-07-12 00:07:13.185890659 +0000
-@@ -424,7 +424,6 @@
-     _, arch = linux.split("_", 1)
-     yield from _manylinux.platform_tags(linux, arch)
-     yield from _musllinux.platform_tags(arch)
--    yield linux
+@@ -498,8 +498,6 @@
+     archs = {"armv8l": ["armv8l", "armv7l"]}.get(arch, [arch])
+     yield from _manylinux.platform_tags(archs)
+     yield from _musllinux.platform_tags(archs)
+-    for arch in archs:
+-        yield f"linux_{arch}"
  
  
  def _generic_platforms() -> Iterator[str]:
diff --git a/sonic-slave-bullseye/disable-non-manylinux.patch b/sonic-slave-bullseye/disable-non-manylinux.patch
index 67e81c4e9116..dcdb953581c9 100644
--- a/sonic-slave-bullseye/disable-non-manylinux.patch
+++ b/sonic-slave-bullseye/disable-non-manylinux.patch
@@ -1,10 +1,16 @@
+Disable any type of wheel besides manylinux wheels. This is to work around an issue where the architecture-specific
+wheel might be built with a newer glibc version than what is supported on the system.
+
+(This patch gets applied only on armhf.)
+
 --- a/tags.py	2022-07-12 00:07:22.710207780 +0000
 +++ b/tags.py	2022-07-12 00:07:13.185890659 +0000
-@@ -424,7 +424,6 @@
-     _, arch = linux.split("_", 1)
-     yield from _manylinux.platform_tags(linux, arch)
-     yield from _musllinux.platform_tags(arch)
--    yield linux
+@@ -498,8 +498,6 @@
+     archs = {"armv8l": ["armv8l", "armv7l"]}.get(arch, [arch])
+     yield from _manylinux.platform_tags(archs)
+     yield from _musllinux.platform_tags(archs)
+-    for arch in archs:
+-        yield f"linux_{arch}"
  
  
  def _generic_platforms() -> Iterator[str]:

From 32e2fc6980fc46e354a917c20e758346be38d090 Mon Sep 17 00:00:00 2001
From: Saikrishna Arcot <sarcot@microsoft.com>
Date: Mon, 8 Jul 2024 17:22:30 -0700
Subject: [PATCH 2/2] Update the pinned version for pip to explicitly point to
 24.1.2

This is needed because the patch file that has been updated can only be
applied to 24.1.2. However, the version pinning for 24.0 will mean that
PR checkers and the daily build will try to continue using 24.0, and
thus fail to apply the patch.

Signed-off-by: Saikrishna Arcot <sarcot@microsoft.com>
---
 files/build/versions/dockers/sonic-slave-bookworm/versions-py3 | 2 +-
 files/build/versions/dockers/sonic-slave-bullseye/versions-py3 | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/files/build/versions/dockers/sonic-slave-bookworm/versions-py3 b/files/build/versions/dockers/sonic-slave-bookworm/versions-py3
index 97db75c96ebf..38d68971250c 100644
--- a/files/build/versions/dockers/sonic-slave-bookworm/versions-py3
+++ b/files/build/versions/dockers/sonic-slave-bookworm/versions-py3
@@ -91,7 +91,7 @@ pep517==0.13.0
 pexpect==4.8.0
 pickleshare==0.7.5
 pillow==10.0.0
-pip==24.0
+pip==24.1.2
 platformdirs==2.6.0
 pluggy==1.0.0+repack
 ply==3.11
diff --git a/files/build/versions/dockers/sonic-slave-bullseye/versions-py3 b/files/build/versions/dockers/sonic-slave-bullseye/versions-py3
index 4bfb6d46bc44..8e9cc4b5d4f4 100644
--- a/files/build/versions/dockers/sonic-slave-bullseye/versions-py3
+++ b/files/build/versions/dockers/sonic-slave-bullseye/versions-py3
@@ -69,7 +69,7 @@ parso==0.8.1
 pexpect==4.8.0
 pickleshare==0.7.5
 pillow==9.4.0
-pip==24.0
+pip==24.1.2
 pluggy==0.13.0
 ply==3.11
 prompt-toolkit==3.0.14
